yt/utilities/linear_interpolators.py

```diff
@@ -1,11 +1,82 @@
+import abc
+
 import numpy as np
 
 import yt.utilities.lib.interpolators as lib
 from yt.funcs import mylog
 
 
-class UnilinearFieldInterpolator:
-    def __init__(self, table, boundaries, field_names, truncate=False):
+class _LinearInterpolator(abc.ABC):
+    _ndim: int
+    _dim_i_type = "int32"
+
+    def __init__(self, table, field_names, truncate=False, *, store_table=True):
+        if store_table:
+            self.table = table.astype("float64")
```

Comment:
this is an implicit data copy. Is it intended ?

Reply:
it was intended in that I was following what the current implementation on main does (but I'm not sure it's necessary).